sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -521,38 +522,96 @@ class Analyzer(
    */
   object ResolveSortReferences extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-      case s @ Sort(ordering, global, p @ Project(projectList, child))
-          if !s.resolved && p.resolved =>
-        val (newOrdering, missing) = resolveAndFindMissing(ordering, p, child)
+      // Here, this rule only resolves the missing sort references if the child is not Aggregate
+      //   Another rule ResolveAggregateFunctions will resolve that case.
```

Comment:
The purpose of that rule is to resolve the aggregate functions (resolving expressions for Sort is kind of side effects), it's better have that in the `ResolveAggregateFunctions`. This also help to reduce the changes in this PR.